File: holoeasy-example-java/src/main/java/org/holoeasy/plugin/MyCounterHolo.java
package org.holoeasy.plugin;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.holoeasy.HoloEasy;
import org.holoeasy.hologram.Hologram;
import org.holoeasy.line.DisplayTextLine;
import org.holoeasy.line.Line;
import org.holoeasy.line.TextLine;
import org.jetbrains.annotations.NotNull;

import java.util.Map;
import java.util.UUID;

public class MyCounterHolo extends Hologram {

    private int clickCount = 0;
    private final Map<UUID, Integer> playerClickCounts = new java.util.HashMap<>();

    private final DisplayTextLine global_counter = displayTextLine(player -> "Clicked " + clickCount + " times")
            .backgroundColor(0x80FF0000)
            .yOffset(1.55f);

    private final TextLine player_counter = textLine(player ->
            "Clicked " + playerClickCounts.getOrDefault(player.getUniqueId(), 0) + " times by " + player.getName())
            .yOffset(-1f);

    private final Line<?> interactionLine = interactionLine()
            .height(1.0f)
            .yOffset(1f);


    public void onClick(@NotNull Player player) {
        clickCount++;
        playerClickCounts.compute(player.getUniqueId(), (uuid, count) -> count == null ? 1 : count + 1);

        global_counter.updateAll();
        player_counter.update(player);
    }

    public MyCounterHolo(@NotNull HoloEasy lib, @NotNull Location location) {
        super(lib, location);
    }

}

File: holoeasy-example-kotlin/src/main/kotlin/org/holoeasy/plugin/MyCounterHolo.kt
package org.holoeasy.plugin

import org.bukkit.Color
import org.bukkit.Location
import org.bukkit.entity.Player
import org.holoeasy.HoloEasy
import org.holoeasy.hologram.Hologram
import java.util.*

class MyCounterHolo( lib: HoloEasy, location: Location) : Hologram(lib, location) {

    private var clickCount = 0
    private val playerClickCounts: MutableMap<UUID, Int> = mutableMapOf()

    private val global_counter = displayTextLine { "Clicked " + (++clickCount) + " times" }
        .backgroundColor(Color.GREEN)

    private val player_counter =
        textLine { "Clicked ${playerClickCounts.getOrDefault(it.uniqueId, 0)} times by ${it.name}" }

    private val interactionLine = interactionLine()


    fun onClick(player: Player) {
        clickCount++
        playerClickCounts.compute(player.uniqueId) { _, count -> if (count == null) 1 else count + 1 }

        global_counter.updateAll()
        player_counter.update(player)
    }
}